core/src/main/java/com/linecorp/armeria/client/endpoint/dns/DnsAddressEndpointGroup.java

```diff
@@ -16,150 +16,183 @@
 
 package com.linecorp.armeria.client.endpoint.dns;
 
-import static com.google.common.base.Preconditions.checkArgument;
-import static com.google.common.base.Preconditions.checkState;
-import static com.google.common.collect.ImmutableList.toImmutableList;
-import static java.util.Objects.requireNonNull;
-
-import java.net.InetAddress;
-import java.time.Duration;
 import java.util.List;
-import java.util.concurrent.TimeUnit;
-
-import javax.annotation.Nullable;
+import java.util.stream.Collectors;
 
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
-
-import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSortedSet;
 
 import com.linecorp.armeria.client.Endpoint;
 import com.linecorp.armeria.client.endpoint.DynamicEndpointGroup;
+import com.linecorp.armeria.client.retry.Backoff;
 import com.linecorp.armeria.common.CommonPools;
 
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.ByteBufHolder;
 import io.netty.channel.EventLoop;
-import io.netty.resolver.NameResolver;
-import io.netty.util.concurrent.Future;
-import io.netty.util.concurrent.ScheduledFuture;
+import io.netty.handler.codec.dns.DefaultDnsQuestion;
+import io.netty.handler.codec.dns.DnsQuestion;
+import io.netty.handler.codec.dns.DnsRawRecord;
+import io.netty.handler.codec.dns.DnsRecord;
+import io.netty.handler.codec.dns.DnsRecordType;
+import io.netty.resolver.ResolvedAddressTypes;
+import io.netty.resolver.dns.DnsServerAddressStreamProvider;
+import io.netty.util.NetUtil;
 
 /**
- * {@link DynamicEndpointGroup} which resolves targets using DNS address queries (A and AAAA). This is useful
- * for environments where service discovery is handled using DNS - for example, Kubernetes uses SkyDNS for
- * service discovery.
+ * {@link DynamicEndpointGroup} which resolves targets using DNS address queries ({@code A} and {@code AAAA}).
```

Comment:
Consider changing to `{@link DnsEndpointGroup} which...`

Reply:
`DnsEndpointGroup` is a package-private class so we can't refer to it here.